### PR TITLE
feat(badge): add `NEW` badge

### PR DESCRIPTION
**Pull request by dancormier**
[STACKS-529](https://stackoverflow.atlassian.net/browse/STACKS-529)

https://deploy-preview-1647--stacks.netlify.app/product/components/badges/#badge-states

---

### Dark mode contrast (as of Tue Feb 13, 2024)

The dark mode version of the "new" badge doesn’t meet our APCA Lc 60 minimum.

<details>
<summary>Terminal output on accessibility test run</summary>

```sh
❌ badge > a11y: s-badge-dark-new should be accessible
      Error: Accessibility Violations
      ---
      Rule: color-contrast-apca-custom
      Impact: serious
      Elements must meet APCA conformance minimums thresholds ( 

Issue target: span
  Context: <span class="s-badge s-badge__new " data-testid="s-badge-dark-new">badge</span>

  Fix all of the following:
    Element has insufficient APCA custom level contrast of 52.46Lc (foreground color: #adb1fa, background color: #363a87, font size: 9.0pt (12px), font weight: 700). Expected minimum APCA lightness contrast of 60Lc
```

</details>

Should we a) alter the badge colors specifically for dark mode, b) alter the colors for both light/dark mode, or c) alter purple values in dark mode so purple-400 and purple 100 result in >= Lc 60?

See also: [Jira comment](https://stackoverflow.atlassian.net/browse/STACKS-529?focusedCommentId=327185)

### CODE REVIEW

**Review by giamir — APPROVED** [review]
Changes look good. I took the liberty of cleaning up unnecessary committed images from the PR diff. https://github.com/StackExchange/Stacks/pull/1647/commits/fbf579d94dc1b285b0fc477856aa3e7bcfabbec3
I hope that's ok. 

I am approving the PR but it would be nice if we could find a fix for the color contrast issue in dark mode instead of skipping the test. I will leave that to you and @CGuindon. 

Thanks and great work as always 🎉

**Review by CGuindon — APPROVED** [review]

**Pull request merged**
